lib/models/lookup.js

```diff
@@ -74,37 +75,45 @@ function LookupModelFactory (Model, assert, Errors, Promise, Constants) {
         setIp: function(ipAddress, macAddress) {
             var self = this;
 
-            return Promise.all([
-                self.findOne({ ipAddress: ipAddress }),
-                self.findOne({ macAddress: macAddress })
-            ]).spread(function(ipRecord, macRecord) {
-                if (macRecord && macRecord.ipAddress === ipAddress) {
-                    throw new Errors.BreakPromiseChainError();
-                }
-                return [ipRecord, macRecord];
-            }).spread(function(ipRecord, macRecord) {
-                if (!ipRecord) {
-                    return [macRecord];
-                }
-                if (ipRecord.ipAddress === ipAddress && ipRecord.macAddress !== macAddress) {
-                    return [
-                        macRecord,
-                        self.destroy({ id: ipRecord.id })
-                            .then(function() {
-                                delete ipRecord.ipAddress;
-                                return self.create(ipRecord);
-                            })
-                    ];
+            var query = {
+                ipAddress: ipAddress, // 10.1.1.2
+                macAddress: { $ne: macAddress } // old mac
+            };
+
+            var update = {
+                $set: {
+                    ipAddress: null
                 }
-            }).spread(function(macRecord) {
-                if (!macRecord) {
-                    return self.create({ ipAddress: ipAddress, macAddress: macAddress });
-                } else {
-                    return self.update({ id: macRecord.id }, { ipAddress: ipAddress });
+            };
+
+            var options = { new: true };
+
+            //Queries for the ipAddress that are not matched with the macAddress
+            //changes ip to null if macAddress doesn't match
+            waterline.lookups.findAndModifyMongo(query, {}, update, options);
```

Comment:
This is async so you'll want to return it otherwise we can't catch errors or guarantee any ordering of this and the findAndModify query below.